includes/phonebook-functions.php

```diff
@@ -0,0 +1,347 @@
+		"organizations",
+		"departments",
+		"staff"
+	) );
+}
+
+/**
+ * Determines if the word 'fax' is in the result name
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $result Object | The phonebook result
+ * @return boolean
+ **/
+function phonebook_filter_fax_in_name( $result ) {
+	return (preg_match( "/^fax\s/i", $result->name ) ||
+		   preg_match( "/\sfax\s/i", $result->name ) ||
+		   preg_match( "/\sfax$/i", $result->name) ) ? false : true;
+}
+
+/**
+ * Formats the result for output to the page.
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $result Array | The result from the search service
+ * @return string | The html output
+ **/
+function format_phonebook_result( $result ) {
+	$is_org = ( $result->from_table === 'organizations' );
+	$is_dept = ( $result->from_table === 'departments' );
+	$is_group = ( $is_org || $is_dept );
+
+	$unique_slug = sanitize_title( str_replace( ' ', '-', $result->name ) . $result->id );
+
+	$class = $is_group ? ' group-result' : '';
+
+	ob_start();
+?>
+	<li class="result<?php echo $class; ?> list-unstyled mb-2">
+		<table class="table">
+			<tbody>
+				<tr class="row mx-auto">
+					<td class="col-sm-6">
+						<?php echo format_phonebook_result_primary( $result, $is_dept, $is_org, $is_group ); ?>
+					</td>
+					<td class="col-sm-3">
+						<?php echo format_phonebook_result_location( $result, $is_dept, $is_org, $is_group ); ?>
+					</td>
+					<td class="col-sm-3">
+						<?php echo format_phonebook_result_contact( $result, $is_dept, $is_org, $is_group ); ?>
+					</td>
+				</tr>
+				<?php if ( ! $is_group && ( ! empty( $result->secondary ) ) ) : ?>
+				<tr class="row mx-auto">
+					<td class="col-sm-12">
+						<a class="person-secondary" data-toggle="collapse" data-target="#<?php echo $unique_slug; ?>"><span class="fa fa-plus"></span> More Results</a>
+						<div class="collapse" id="<?php echo $unique_slug; ?>">
+							<ul class="list-unstyled">
+							<?php foreach( $result->secondary as $secondary ) : ?>
+								<li>
+									<table class="table">
+										<tbody>
+											<tr>
+												<td class="col-sm-6">
+													<?php echo format_phonebook_result_primary( $secondary, false, false, false ); ?>
+												</td>
+												<td class="col-sm-3">
+													<?php echo format_phonebook_result_location( $secondary, false, false, false ); ?>
+												</td>
+												<td class="col-sm-3">
+													<?php echo format_phonebook_result_contact( $secondary, false, false, false ); ?>
+												</td>
+											</tr>
+										</tbody>
+									</table>
+								</li>
+							<?php endforeach; ?>
+							</ul>
+						</div>
+					</td>
+				</tr>
+				<?php elseif ( $is_group && ( count( $result->staff ) > 0 ) ) : ?>
+				<tr class="row mx-auto">
+					<td colspan="3" class="col-sm-12">
+						<a class="toggle" data-toggle="collapse" data-target="#<?php echo $unique_slug; ?>" ><span class="fa fa-plus"></span> Show Staff</a>
```

Comment:
- Icon needs aria-hidden="true" attribute.
- .toggle should have a pointer cursor on hover to help it look more like a clickable element
- Having the icon toggle between an open and closed state would be a helpful ux improvement--you could do a super-simple toggle effect by adding a .fa-minus icon next to the plus icon, then show/hide each icon based on whether the parent .toggle link is collapsed, e.g.
```
.toggle {
  &:not(.collapsed) {
    .fa-plus {
      display: none;
    }
  }

  &.collapsed {
    .fa-minus {
      display: none;
    }
  }
}
```
```
<a class="toggle collapsed" role="button" aria-expanded="false" data-toggle="collapse" data-target="#<?php echo $unique_slug; ?>" >
  <span class="fa fa-plus" aria-hidden="true"></span>
  <span class="fa fa-minus" aria-hidden="true"></span> 
  Show Staff
</a>
```